QuickOSM/core/query_factory.py

```diff
@@ -256,21 +256,76 @@ def generate_xml(self) -> str:
 
         return query
 
-    def make(self) -> str:
+    def generate_oql(self) -> str:
```

Comment:
I'm thinking if we should keep a single `generate` function and make the difference between OQL and XML only when editing the `query` string.
* So we remove duplicated code
* Some logic is only coded once `enumerate`, `len`, `for` loops etc
* The problem, is that XML or OQL might get not synced overtime. Maybe the XML function will be updated later and not the OQL one. So if we keep a single function, then it's like required to update both version at the same time.
But I agree, it will make the code less readable.

Reply:
Ok, let's keep it like this.